adapter/pkg/messaging/azure_listener.go

```diff
@@ -0,0 +1,160 @@
+		subManager, err := ns.NewSubscriptionManager(topicName)
+
+		if err != nil {
+			logger.LoggerMgw.Error("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error occurred while trying get subscription" +
+				" manager from azure service bus for topic name " + topicName + ":%v", err)
+		}
+
+		ctx, cancel := context.WithCancel(parentContext)
+		defer cancel()
+		availableSubscriptionList, err := subManager.List(ctx)
+
+		if err != nil {
+			logger.LoggerMgw.Error("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error occurred while trying get available " +
+				"subscriptions from azure service bus for " + "topic name " + topicName + ":%v", err)
+		}
+
+		if isSubscriptionExist(subscriptionName, availableSubscriptionList) {
+			logger.LoggerMgw.Debugf("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Subscription " + subscriptionName +
+				"exist in the system")
+			subscriptionExistForFurtherProcess = true
+		} else {
+			ctx, cancel := context.WithCancel(parentContext)
+			defer cancel()
+			_, err = subManager.Put(ctx, subscriptionName)
+
+			if err != nil {
+				logger.LoggerMgw.Error("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error occurred while trying to create " +
+					"subscription " + subscriptionName + " from azure service bus for topic name " +
+						topicName + ":%v", err)
+			} else {
+				logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Subscription " +
+					subscriptionName + "created")
+				subscriptionExistForFurtherProcess = true
+			}
+		}
+
+		if subscriptionExistForFurtherProcess {
+			logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] subscription " + subscriptionName +
+				" exist. Can proceed")
+			//topic subscription client creation
+			topicSubscriptionClient, err := subManager.Topic.NewSubscription(subscriptionName)
+
+			if err != nil {
+				logger.LoggerMgw.Error("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error occurred while trying to create " +
+					"topic subscription client for  " + subscriptionName + " from azure service bus for topic name " +
+						topicName + ":%v", err)
+			} else {
+				dataChannel := make(chan []byte)
+				ackChannel := make(chan bool)
+				if strings.EqualFold(topicName, notification) {
+					dataChannel = AzureNotificationChannel
+					ackChannel = AzureNotificationAck
+				} else if strings.EqualFold(topicName, tokenRevocation) {
+					dataChannel = AzureRevokedTokenChannel
+					ackChannel = AzureRevokedTokenAck
+				}
+
+				logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Starting to receive messages for " +
+					"subscriptionName  " + subscriptionName + " from azure service bus for topic name " + topicName)
+
+				ctx, cancel := context.WithCancel(parentContext)
+				defer cancel()
+				err = topicSubscriptionClient.Receive(ctx, servicebus.HandlerFunc(func(ctx context.Context,
+					message *servicebus.Message) error {
+					dataChannel <- message.Data
+					<-ackChannel
```

Comment:
The process will halt if an error occurs in channel consuming part it seems. As the there is a continue statement within loop prior to sending an Ack through the channel.

Reply:
Ya true this is just to test till event consumption. Basically we need to <-ackChannel before doing continue also . I kept it till i finish other logic but will add a todo